src/nvim/ex_cmds.c

```diff
@@ -2946,6 +2951,16 @@ void do_sub(exarg_T *eap)
   int start_nsubs;
   int save_ma = 0;
 
+  sub_done = 0;
+  bool last_is_slash = false;
+  char_u *c = eap->arg;
+
+  // checking if the last char of the cmdline is a '/'
+  if (*c != '\0') {
+    do { c++; } while (*c != '\0');
+    last_is_slash = (*(c-1) == '/');
```

Comment:
`c` is not needed. Cycle is also not needed. Even `if` may be avoided, though somebody will probably say `if` is more readable:

```C
const bool last_is_slash = !(eap->arg == NUL
                             || eap->arg[STRLEN(eap->arg) - 2] != '/'`);
```